bpo-40246: Revert reporting of invalid string prefixes (#19888)
python/cpython: Misc/NEWS.d/next/Core and Builtins/2020-05-03-23-28-11.bpo-40246.c1D7x8.rst

> Reporting a specialised error message for invalid string prefixes, which was introduced in #19888, is being reverted due to backwards compatibility concerns for strings that immediately follow a reserved keyword without whitespace between them. Constructs like `bg="#d00" if clear else"#fca"` were failing to parse, which is not an acceptable breakage on such short notice.

Reviewer: 19888 is this one

Author: Whoops. Thanks for the catch!

> Reporting a specialised error message for invalid string prefixes, which was introduced in GH-19476, is being reverted due to backwards compatibility concerns for strings that immediately follow a reserved keyword without whitespace between them. Constructs like `bg="#d00" if clear else"#fca"` were failing to parse, which is not an acceptable breakage on such short notice.

Reviewer 2: Please, mention the bpo number instead of the PR number (GH-19476)

> Reporting a specialised error message for invalid string prefixes, which was introduced in GH-19476, is being reverted due to backwards compatibility concerns for strings that immediately follow a reserved keyword without whitespace between them. Constructs like `bg="#d00" if clear else"#fca"` were failing to parse, which is not an acceptable breakage on such short notice.

Reviewer 2: ```suggestion
Reporting a specialised error message for invalid string prefixes, which was introduced in GH-19476, is being reverted due to backwards compatibility concerns for strings that immediately follow a reserved keyword without whitespace between them, identifying constructs like `bg="#d00" if clear else"#fca"` incorrectly invalid.
```

Reviewer 3: Honestly I like @lysnikolaou's version better.

Reviewer 2: Fair enough, let's skip my suggestion then.

> Reporting a specialised error message for invalid string prefixes, which was introduced in GH-19476, is being reverted due to backwards compatibility concerns for strings that immediately follow a reserved keyword without whitespace between them. Constructs like `bg="#d00" if clear else"#fca"` were failing to parse, which is not an acceptable breakage on such short notice.

Reviewer 2: ```suggestion
Reporting a specialised error message for invalid string prefixes, which was introduced in :issue:`40246`, is being reverted due to backwards compatibility concerns for strings that immediately follow a reserved keyword without whitespace between them. Constructs like `bg="#d00" if clear else"#fca"` were failing to parse, which is not an acceptable breakage on such short notice.
```
